lexicon/providers/dreamhost.py

```diff
@@ -0,0 +1,181 @@
+
+import requests
+from lexicon.providers.base import Provider as BaseProvider
+
+
+LOGGER = logging.getLogger(__name__)
+
+NAMESERVER_DOMAINS = ['dreamhost.com']
+
+
+def provider_parser(subparser):
+    """Module provider for Linode"""
+    subparser.add_argument(
+        "--auth-token", help="specify api key for authentication")
+
+
+class Provider(BaseProvider):
+    """Provider class for Linode"""
+
+    def __init__(self, config):
+        super(Provider, self).__init__(config)
+        self.domain_id = None
+        self.api_endpoint = 'https://api.dreamhost.com/'
+
+    def _authenticate(self):
+        self.domain_id = None
+        payload = self._get('domain-list_domains')
+        data = payload.get('data', None)
+        if data is None:
+            raise Exception('Domain not found')
+
+        for domain in data:
+            if domain.get('domain', '') == self.domain:
+                self.domain_id = self.domain
+        if self.domain_id is None:
+            raise Exception('Domain not found')
+
+    def _create_record(self, rtype, name, content):
+        name = self._full_name(name)
+        payload = self._get('dns-add_record', query_params={
+            'record': name,
+            'type': rtype,
+            'value': content,
+        })
+
+        if (payload.get('result', '') != 'success' and
+                payload.get('data', '') != 'record_already_exists_remove_first' and
+                payload.get('data', '') != 'record_already_exists_not_editable'):
+            raise Exception('unable to add record: %s' % payload)
+
+        return True
+
+    # List all records. Return an empty list if no records found
+    # type, name and content are used to filter records.
+    # If possible filter during the query, otherwise filter after response is received.
+    def _list_records(self, rtype=None, name=None, content=None):
+
+        payload = self._get('dns-list_records')
+
+        if payload.get('result', '') != 'success':
+            raise Exception('unable to get records: %s' % payload)
+
+        resource_list = payload.get('data', None)
+        if resource_list is None:
+            raise Exception('unable to get records: %s' % payload)
+
+        resource_list = [
+            resource for resource in resource_list if resource['zone'] == self.domain]
+        if rtype:
+            resource_list = [
+                resource for resource in resource_list if resource['type'] == rtype]
+        if name:
+            name = self._full_name(name)
+            resource_list = [
+                resource for resource in resource_list if resource['record'] == name]
+        if content:
+            resource_list = [
+                resource for resource in resource_list if resource['value'] == content]
+
+        processed_records = []
+        for resource in resource_list:
+            processed_records.append({
+                'id': resource['type'] + '|' + resource['record'] + '|' + resource['value'],
```

Comment:
If you want, you can reuse the method `_identifier` from godaddy provider, to build a more ID-like value here.

Reply:
I'll wrap with Provider._identifier to make the code cleaner.

However, as explained in the later sections, I think I'll use json.dumps as the core-function in Provider._identifier.